api/server/server.go

```diff
@@ -203,7 +203,7 @@ func postContainersKill(eng *engine.Engine, version version.Version, w http.Resp
 	name := vars["name"]
 
 	// If we have a signal, look at it. Otherwise, do nothing
-	if sigStr := vars["signal"]; sigStr != "" {
+	if sigStr := r.Form.Get("signal"); sigStr != "" {
```

Comment:
Correct, when testing the fix, a patch came in that blew up our fix.  So I included it in the pull request.